joplin/api/schema.py

```diff
@@ -6,19 +6,104 @@
+
+@convert_django_field.register(RichTextField)
+def convert_rich_text_field(field, registry=None):
+    return RichTextFieldType(
+        description=field.help_text, required=not field.null
+    )
 
 
 class StreamFieldType(Scalar):
+    """
+    todo
+            for item in serialized:
+                loop through all the values and try running expand_db_html on them to return an href
+                still need to find a way to actually attach the janis_url to the selection tho..
+    """
+
     @staticmethod
-    def serialize(dt):
-        return [{'type': item.block_type, 'value': item.block.get_api_representation(item.value), 'id': item.id} for item in dt]
+    def serialize(StreamValue):
+        """
+        This is a rats nest of loops and conditionals, but the general idea is two functions:
+            1)loop through all the objects in the StreamValue, and
+                if it is a RichTextBlock,
+                add its name to a list of keys
+                ( this dosen't quite capture all the use cases as needed so it needs work)
+            2) use each of the above names as keys to sort through the stream_data
+                (which is a python dict represention of the StreamField, alot like JSON)
+                use the nested_lookup to recursivley return a list of all values that match the key
+                run a nested_alter using expand_db_html as a callback to edit those values in stream_data
+                (otherwise the values tend to get overwritten)
+
+        """
+        print(type(StreamValue))
+        try:
+            data = StreamValue.stream_data
+            rt_names = []
+            rt_names.append('options')
+            # get lists of potential keys
+            try:
+                for item in StreamValue:
+                    block = item.block
+                    try:
+                        if isinstance(block, RichTextBlock):
+                            rt_names.append(block.name)
+                        elif block.child_blocks:
+                            for child_block in block.all_blocks():
+                                if isinstance(child_block, RichTextBlock):
+                                    rt_names.append(child_block.name)
+                    except AttributeError as e:
+                        pass
+
+                # todo: make things less nested
+                for block_key in rt_names:
+                    print(block_key)
+                    values = nested_lookup(block_key, data)
+
+                    for value in values:
+                        if not isinstance(value, str):
+                            for elem in value:
+                                for key in rt_names:
+                                    print('alter elem', key)
+                                    altered_value = nested_alter(elem, key, expand_db_html)
+                # fallback to 'value'
+                for elem in data:
+                    if elem['type'] == 'basic_step':
```

Comment:
Alot of the initial attempt at this was to have a function that dynamically got all the cases needed from our code to alter, so that we wouldn't have to revisit every time we do something fancy with StreamFields. 
However it was pretty hard to get this working due to the complex data structures, so I ended up hard-coding in some use cases to make it functional. 
So now its kind of a mix of generally reusable stuff and hard-coded stuff, which is...not great.